### ISSUE TITLE
Make paragraph style arguments always precede text style arguments.

### ISSUE THREAD

**Issue by TomasMikula** [TomasMikula/RichTextFX#245]
Since paragraph style is applied on a higher level than text style, paragraph style arguments should precede text style arguments. We should make this consistent throughout the code base. This applies to both type parameters and constructor/method parameters. We have had this wrong in most places since the very introduction of paragraph-level styles. Affected classes include `StyledTextArea` and its subclasses and super-interfaces, `StyledTextAreaModel`, `StyledDocument` and its subclasses, `Paragraph`.

For example,

```java
StyledTextArea<S, PS>
```

should be changed to

```java
StyledTextArea<PS, S>
```

and the constructor

```java
StyledTextArea(
        initialStyle, applyStyle,
        initialParagraphStyle, applyParagraphStyle
)
```

should be changed to

```java
StyledTextArea(
        initialParagraphStyle, applyParagraphStyle,
        initialStyle, applyStyle
)
```

**Comment by TomasMikula** [comment]
There are already going to be many backwards incompatible changes in 0.7, so that is a good chance to also get in this one.

**Comment by TomasMikula** [comment]
Resolved by PR #246.
